src/Proj4.jl

```diff
@@ -125,8 +124,8 @@ function transform!(src::Projection, dest::Projection, position::Array{Float64,2
     P = pointer(position)
     x = P
     y = P + sizeof(Cdouble)*npoints
-    z = ncomps < 3 ? C_NULL : P + 2*sizeof(Cdouble)*npoints
-    err = _transform!(src, dest, npoints, 1, x, y, z)
+    z = (ncomps < 3) ? Ptr{Float64}(C_NULL) : P + 2*sizeof(Cdouble)*npoints
+    err = _transform!(src, dest, npoints, Compat.@compat(Int32(1)), x, y, z)
```

Comment:
If you avoid overspecifying the types above, you can avoid this cast.  I think...